examples/fairmotion_interface.py

```diff
@@ -0,0 +1,478 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import json
+import os
+from enum import Enum
+from typing import Any, Dict, List, Optional, Tuple
+
+import magnum as mn
+from fairmotion.core import motion
+from fairmotion.data import amass
+from fairmotion.ops import conversions
+
+import habitat_sim
+from habitat_sim.logging import LoggingContext, logger
+from habitat_sim.physics import JointType
+
+#### Constants
+ROOT = 0
+FILE_SUFFIX = "_fm_data"
+METADATA_DEFAULT = {
+    "default": {
+        "urdf_path": "../habitat-sim/data/test_assets/urdf/amass_male.urdf",
+        "amass_path": "../fairmotion/amass_test_data/CMU/CMU/06/06_12_poses.npz",
+        "bm_path": "../fairmotion/amass_test_data/smplh/male/model.npz",
+        "rotation": mn.Quaternion.rotation(mn.Deg(-90), mn.Vector3.x_axis())
+        * mn.Quaternion.rotation(mn.Deg(90), mn.Vector3.z_axis()),
+        "translation": mn.Vector3([2.5, 0.0, 0.7]),
+    }
+}
+METADATA_DIR = "../habitat-sim/data/fairmotion/"
+
+
+class FairmotionInterface:
+    def __init__(
+        self,
+        sim,
+        metadata_name=None,
+        urdf_path=None,
+        amass_path=None,
+        bm_path=None,
+        metadata_dir=None,
+    ) -> None:
+        LoggingContext.reinitialize_from_env()
+        self.sim = sim
+        self.art_obj_mgr = self.sim.get_articulated_object_manager()
+        self.model: habitat_sim.physics.ManagedArticulatedObject = None
+        self.motion: motion.Motion = None
+        self.metadata = {}
+        self.metadata_dir = metadata_dir or METADATA_DIR
+        self.motion_stepper = 0
+        self.setup_default_metadata()
+
+        self.key_frames = None
+        self.key_frame_models = []
+        self.preview_mode = Preview.OFF
+        self.traj_ids: List[int] = []
+        self.is_reversed = False
+
+        # loading from file if given
+        # if file doesn't exist, make a new set of data with
+        # metadata_name as name
+        if metadata_name is not None:
+            try:
+                self.fetch_metadata(metadata_name)
+                self.set_data(name=metadata_name)
+            except Exception:
+                Exception(f"No file with name {metadata_name}, creating new file.")
```

Comment:
@JuanTheEngineer @aclegg3 This is a bug. It does nothing. If you intended to raise it, the code after it it in the exception block wouldn't run. Did you mean to make this a logger.err?